docs/decisions/0003-theming-using-webpack-module-federation.rst

```diff
@@ -0,0 +1,125 @@
+with the MFE, allowing the current mechanisms of theming to continue working as
+they do now.
+
+Technical Details
+-----------------
+
+As the module federation feature is currently only available in Webpack 5.
+The first step is to upgrade ``frontend-build`` to use the latest Webpack 5
+release.
+
+Additionally, ``frontend-build`` can also automatically include the Webpack
+config to mark the theme as sharable, meanwhile ``frontend-platform`` can
+include the common code for the theme, so it doesn't need to be included in each
+MFE.
+
+To allow each MFE to have its own SCSS that's loaded, and to have control over
+the order, the parts of the theme to be loaded before the regular theme and
+after it can be specified during app initialization.
+
+Let us for example look at the ``index.scss`` file of ``frontend-app-account``.
+In this file we have an `initial segment that is common across all MFEs`_ and
+`the rest`_ is code specific to the MFE. The initial segment will now move to
+``frontend-platform``.
+
+The ``frontend-platform`` code will attempt to load the dynamic theme and fall
+back to the standard theme, and also load the custom scss from the app.
+
+.. _initial segment that is common across all MFEs: https://github.com/edx/frontend-app-account/blob/e2f9edd62317de2d762e87f7f68d58b31aab3497/src/index.scss#L1-L9
+.. _the rest: https://github.com/edx/frontend-app-account/blob/e2f9edd62317de2d762e87f7f68d58b31aab3497/src/index.scss#L11-L64
+
+During compilation all MFEs will now support a ``THEME_URL``. If provided this
+is where the theme will be loaded from dynamically by ``frontend-platform``.
+
+To avoid a flash of  content without styling, ``frontend-platform`` can add a lifecycle
+phase for theming that will block triggering the ``APP_READY`` signal till the
+theme is loaded.
+
+Consequences
+------------
+
+The theme will no longer be included as a standard SCSS file that's included in
+the base html file for the MFE, instead it will be loaded dynamically from the
+configured theme MFE URL.
+
+Since the theme is now loaded from an external source and uses a standard
+Webpack pluggable format, third parties can build any mechanism they like for
+updating this link with new theme as long as it follows the format Webpack
+expects for modules.
+
+One downside of this mechanism is that since the theme is no longer built as
+part of each MFE, there is no way currently, to use Paragon/bootstrap variables
+in custom SCSS used by an MFE.
+
+For example, `this usage of $grid-breakpoints`_ in an MFE component's style will
+no longer work. While it will be possible for the file to compile. It will no
+longer respect changes to that variable in the base theme, since that will only
+work during SCSS compilation.
+
+The solution for the above is to avoid using such variables in themes directly,
+and instead of rely on Bootstrap/Paragon classes to apply the desired effect.
```

Comment:
nit: 
Can be changed to `... themes directly and instead, rely on ...` or `...themes directly, and rely on ...`